### DISCUSSION TITLE
Hello all.  Status bgc support?

### DISCUSSION THREAD

**Discussion by gmaze** [euroargodev/argopy#254] [General]
Hello all.  Status bgc support?  No errors for dataset='phy'

```
argopy.set_options(src='gdac', dataset='bgc', mode='expert')
ArgoSet = ArgoDataFetcher().region(region_date)
df = ArgoSet.to_dataframe()

```
returns error in fetchers.py:556

ValueError: Errors happened with all URLs, this could be due to an internal impossibility to read returned content.

_Originally posted by @SBS-EREHM in https://github.com/euroargodev/argopy/issues/202#issuecomment-1496850959_

**Comment by gmaze** [comment]
@SBS-EREHM As you can see ton #202 we're working hard to provide BGC support, but it's not here yet !
It's difficult to put a timeline on BGC data availability

I expect partial support before this summer, *partial* meaning not all variables and data sources will be supported at the beginning, we'll go gradually

**Reply by SBS-EREHM** [reply]
Thanks @gmaze.

I am working on regional queries of CDOM over the complete time of BGC deployments (2012 - present).  I can use **[argovis](https://github.com/argovis/demo_notebooks)** for now (see Exploring_Argo_BGC_with_Argovis.{ipynb, py}), but it's slow and (from my perspective) retrieving profile data obscure, but the initial query is similar:

```
from argovisHelpers import helpers as avh

params = {
        'startDate': '2012-01-01T00:00:00Z',
        'endDate':   '2013-01-01T00:00:00Z',
        'source': 'argo_bgc',
        'polygon': polygon,
        'data': 'all'
    }

d  = avh.query('argo', options=params, apikey=API_KEY, apiroot=API_PREFIX)
```

**Comment by gmaze** [comment]
I forget to post an update here 😨 

**argopy** started to provided [BGC support after release v0.1.14](https://argopy.readthedocs.io/en/latest/whats-new.html#v0-1-14-29-sep-2023) !

We [keep working on improving BGC support](https://x.com/mazeguillaume/status/1797909281595179322), most notably to implement easy parameter data mode filtering for the "Standard" user mode
